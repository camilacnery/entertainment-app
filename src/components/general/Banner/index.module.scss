@use '@/styles/constants' as *;

.banner {
  opacity: 0.5;
  z-index: 1;
  position: absolute;

  & img {
    object-fit: cover;
    width: 100vw;
    height: 100vh;
  }

  &:before,
  &:after {
    content: "";
    width: 100%;
    height: 50vh;
    position: absolute;
    z-index: 1;
  }

  &:before {
    top: 0;
    background-image: linear-gradient(to top, rgba(0, 0, 0, 0), rgba(13, 13, 13, 0.63) 52%, $background-color);
  }

  &:after {
    bottom: 0;
    left: 0;
    background-image: linear-gradient(to bottom, rgba(0, 0, 0, 0), rgba(21, 21, 22, 0.4) 28%, $background-color 100%);
  }
}