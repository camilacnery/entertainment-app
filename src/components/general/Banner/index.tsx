import { FC } from "react";
import Image from "next/future/image";
import styles from "./index.module.scss";

type TBanner = {
  imageUrl: string;
};

const Banner: FC<TBanner> = ({ imageUrl }) => {
  return (
    <div className={styles.banner}>
      <Image src={imageUrl} alt={""} width={1024} height={768} />
    </div>
  );
};

export default Banner;
