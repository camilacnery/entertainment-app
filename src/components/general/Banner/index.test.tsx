import { render } from "@testing-library/react";

import Banner from ".";

describe("Banner", () => {
  it("renders component", () => {
    const { baseElement } = render(<Banner imageUrl="https://image.url" />);

    expect(baseElement).toMatchSnapshot();
  });
});
