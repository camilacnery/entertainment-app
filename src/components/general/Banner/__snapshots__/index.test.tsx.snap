// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Banner renders component 1`] = `
<body>
  <div>
    <div
      class="banner"
    >
      <img
        alt=""
        data-nimg="future"
        decoding="async"
        height="768"
        loading="lazy"
        src="/_next/image?url=https%3A%2F%2Fimage.url&w=2048&q=75"
        srcset="/_next/image?url=https%3A%2F%2Fimage.url&w=1080&q=75 1x, /_next/image?url=https%3A%2F%2Fimage.url&w=2048&q=75 2x"
        style="color: transparent;"
        width="1024"
      />
    </div>
  </div>
</body>
`;
